### PR TITLE
Add support for scheduled job

### PR DESCRIPTION
**Pull request by more-ron**
**Feature:**
This will add support for scheduled jobs.

**Ticket:**
Mainly for [MRR + ARPPU](https://app.asana.com/0/41592882922934/39842033097283) but could be used for precise data gathering time for [Analytics for lates](https://app.asana.com/0/41592882922934/36107421278586)

**Submitter Checklist:**
- [x] CI Passing
- [x] PR Linked to Asana Ticket
- [x] Hound Appeased Acceptably
- [x] No Code Climate Issues
- [x] Class, Module, and public method comments up to spec

**Reviewer Checklist:**
- [ ] New Code Tested Effectively
- [ ] Code organization meets SOP standards
- [ ] Does this code fulfill the ticket?

**Only Then:**
- [ ] Move ticket in Asana

### CODE REVIEW

**Comment by more-ron** [comment]
Do not merge yet. Needs https://github.com/payrollhero/dispatch-rider/pull/68

File: Gemfile [hunk 1]
```diff
@@ -2,3 +2,5 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in activejob-dispatch_rider.gemspec
 gemspec
+
+gem "dispatch-rider", github: "payrollhero/dispatch-rider", ref: "9120a4e14217104d65ca9ddd41af138bf66dfb93"
```
Comment (piotrb) [inline Gemfile]: hmm .. that is not supposed to be here is it?

**Comment by piotrb** [comment]
Looks reasonable

File: lib/activejob/queue_adapters/dispatch_rider_adapter.rb [hunk 2]
```diff
@@ -5,18 +5,15 @@ class DispatchRiderAdapter
...
-          raise NotImplementedError,
-                "This queueing backend does not support scheduling jobs. To "\
-                "see what features are supported go to "\
-                "http://api.rubyonrails.org/classes/ActiveJob/QueueAdapters.html"
+        # @param [ActiveJob::Base] job
+        # @param [Float] scheduled_at - float representing time (blame AJ)
+        def enqueue_at(job, scheduled_at)
+          scheduled_job_details = job_details_for(job).merge(scheduled_at: Time.at(scheduled_at))
+
+          ::DispatchRider::ScheduledJob.create! scheduled_job_details
```
Comment (more-ron) [inline lib/activejob/queue_adapters/dispatch_rider_adapter.rb:16]: This just add support for `#enqueue_at`.

File: Gemfile (hunk 1 above)
Comment (more-ron) [inline Gemfile]: Yup. That's why there's a note to not merge. This is away to include it in the gem before it's released. Unfortunately CircleCI is not able to load it so it only works locally.

**Pull request merged**